+adi/+FMComms5/Rx.m

```diff
@@ -0,0 +1,311 @@
+                id = 'voltage0';
+                obj.setAttributeBool(id,'rf_dc_offset_tracking_en',value,false,obj.iioDevPHYChipB); %#ok<MCSUP>
+            end
+        end
+        % Check EnableRFDCTracking
+        function set.EnableBasebandDCTrackingChipB(obj, value)
+            obj.EnableBasebandDCTrackingChipB = value;
+            if obj.ConnectedToDevice
+                id = 'voltage0';
+                obj.setAttributeBool(id,'bb_dc_offset_tracking_en',value,false,obj.iioDevPHYChipB); %#ok<MCSUP>
+            end
+        end
+        % Check CenterFrequency
+        function set.CenterFrequencyChipB(obj, value)
+            validateattributes( value, { 'double','single' }, ...
+                { 'real', 'positive','scalar', 'finite', 'nonnan', 'nonempty','integer','>=',70e6,'<=',6e9}, ...
+                '', 'CenterFrequency');
+
+            obj.CenterFrequencyChipB = value;
+            if obj.ConnectedToDevice
+                id = sprintf('altvoltage%d',strcmp(obj.Type,'Tx'));
+                obj.setAttributeLongLong(id,'frequency',value,true,4,obj.iioDevPHYChipB); %#ok<MCSUP>
+            end
+        end
+        % Check RFBandwidth
+        function set.RFBandwidthChipB(obj, value)
+            validateattributes( value, { 'double','single' }, ...
+                { 'real', 'positive','scalar', 'finite', 'nonnan', 'nonempty','integer','>=',200e3,'<=',56e6}, ...
+                '', 'RFBandwidth');
+            
+            obj.RFBandwidthChipB = value;
+            if obj.ConnectedToDevice && ~obj.EnableCustomFilter
+                id = 'voltage0';
+                obj.setAttributeLongLong(id,'rf_bandwidth',value,strcmp(obj.Type,'Tx'),30,obj.iioDevPHYChipB); %#ok<MCSUP>
+            end
+        end
+        function set.LoopbackModeChipB(obj, value)
+            validateattributes( value, { 'double','single', 'uint32' }, ...
+                { 'real', 'nonnegative','scalar', 'finite', 'nonnan', 'nonempty','integer','>=',0,'<=',2}, ...
+                '', 'LoopbackMode');    
+            obj.LoopbackModeChipB = value;
+            if obj.ConnectedToDevice
+                obj.setDebugAttributeLongLong('loopback',value,1,obj.iioDevPHYChipB); %#ok<MCSUP>                    
+            end
+        end
+    end
+    
+    %% API Functions
+    methods (Hidden, Access = protected)
+        function setupInit(obj)
+            % Write all attributes to device once connected through set
+            % methods
+            setupLibad9361(obj);
+            obj.iioDevPHYChipB = calllib('libiio', 'iio_context_find_device',obj.iioCtx,'ad9361-phy-B');
+            % Do writes directly to hardware without using set methods.
+            % This is required sine Simulink support doesn't support
+            % modification to nontunable variables at SetupImpl
+            
+            % Gains
+            obj.setAttributeRAW('voltage0','gain_control_mode',obj.GainControlModeChannel0,false);
+            if obj.channelCount>2
```

Comment:
Might not want to use channelCount here. If EnabledChannels == [3,4] the gain modes will not be updated

Reply:
Addressed by the commit.